src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -408,6 +407,24 @@ internal ResultVersionOne CreateResult(Result v2Result)
                     }
                 }
 
+                if (_currentV2Run.Resources?.Rules != null)
+                {
```

Comment:
Bug: If there's no rules dictionary, you'll neglect to populate v1's `result.ruleId`.

Reply:
In that case, just copy over v2's `result.ruleId`.

Reply:
Fixed